packages/server-wallet/src/engine/engine.ts

```diff
@@ -548,14 +549,19 @@ export class SingleThreadedEngine
     const objectives = await this.store.getObjectives(channelIds);
 
     await Promise.all(
-      objectives.map(
-        async ({type, objectiveId}) =>
-          type === 'OpenChannel' && (await this.store.approveObjective(objectiveId))
-      )
+      objectives
+        .filter(o => o.type === 'OpenChannel')
+        .map(async ({objectiveId}) => await this.store.approveObjective(objectiveId))
     );
 
     await this.takeActions(channelIds, response);
 
+    for (const channelId of channelIds) {
+      if (this.richObjectives[channelId]) {
+        await this.crankRichObjective(channelId, {type: 'Nudge', now: 0}, response);
```

Comment:
It represents sending an event to the machine Perhaps it'd be better to call it `'Crank'`?

Since the protocol is a pure function that is meant to "handle timeouts". So, you might want to nudge it every now and then to see if it's timed out.

Reply:
Got it! I would prefer sticking with the existing "crank" terminology.

Reply:
7341e77fb2a9969e8cbd68cea6917eba8d72b365